bode/bode/models/task_relations_actions.py

```diff
@@ -51,6 +54,54 @@ def get_relation_types(task_id):
     return {map_to_related_task_schema(relation, task_id) for relation in relations_lhs + relations_rhs}
 
 
+def get_transitive_interchangable_related_tasks(task_id, filters=list()):
+    from bode.models.task import Task
+
+    def get_distinct_uuids(uuids):
+        return list(set(uuids))
+
+    def get_relation_graph(relations):
+        graph = Graph(len(all_interchangable_tasks))
+        uuids = []
+
+        for relation in relations:
+            first_task_id, second_task_id = relation.first_task_id, relation.second_task_id
+            graph.add_edge(str(first_task_id), str(second_task_id))
+            uuids.append(str(first_task_id))
+            uuids.append(str(second_task_id))
+
+        return graph, uuids
+
+    def group_tasks_subsets():
+        task_parent = parents[task_id] if task_id in parents.keys() and parents[task_id] != -1 else task_id
+        tasks_subset = []
+        for key in parents.keys():
+            if task_parent == graph.find_union.find(key):
+                tasks_subset.append(key)
+        return tasks_subset
+
+    all_interchangable_tasks = db.session.query(TaskRelation).filter(*filters).all()
+
+    graph, uuids = get_relation_graph(all_interchangable_tasks)
+    parents = graph.union_subsets(get_distinct_uuids(uuids))
+    interchangable_tasks = group_tasks_subsets()
+
+    return (
+        db.session.query(TaskRelation, Task)
+        .filter(*filters)
+        .filter(
+            or_(
+                TaskRelation.second_task_id.in_(interchangable_tasks),
+                TaskRelation.first_task_id.in_(interchangable_tasks),
+            )
+        )
+        .join(Task, TaskRelation.second_task_id == Task.id)
+        .filter(Task.id != task_id)
+        .distinct(Task.title)
```

Comment:
Doing distinct over `title` might skip some results if there were 2 tasks with the same title in the DB. Any reason why `id` wouldn't work?
```suggestion
        .distinct(Task.id)
```